unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/ExploreV2/Scripts/Sections/PlacesAndEventsSection/SubSections/EventsSubSection/EventsSubSectionMenu/PlaceAndEventsCardsReloader.cs

```diff
@@ -0,0 +1,83 @@
+﻿using DCL;
+using System;
+using UnityEngine;
+
+public class PlaceAndEventsCardsReloader : IDisposable
+{
+    private readonly IPlacesAndEventsSubSectionComponentView view;
+    private readonly DataStore_ExploreV2 exploreV2Menu;
+
+    private readonly Action requestAllFromAPI;
+
+    internal bool firstLoading = true;
+    internal bool reloadSubSection;
+    private bool isWaitingAnimTransition;
+
+    internal float lastTimeAPIChecked;
+
+    public PlaceAndEventsCardsReloader(IPlacesAndEventsSubSectionComponentView view, DataStore_ExploreV2 exploreV2Menu, Action requestAllFromAPI)
+    {
+        this.view = view;
+        this.exploreV2Menu = exploreV2Menu;
+
+        this.requestAllFromAPI = requestAllFromAPI;
+    }
+
+    public void Initialize()
+    {
+        exploreV2Menu.isOpen.OnChange += OnExploreV2Open;
+    }
+
+    public void Dispose()
+    {
+        exploreV2Menu.isOpen.OnChange -= OnExploreV2Open;
+
+        if (isWaitingAnimTransition)
+            exploreV2Menu.isInShowAnimationTransiton.OnChange -= OnAnimationTransitionFinished;
+    }
+
+    internal void OnExploreV2Open(bool current, bool _)
+    {
+        if (!current)
+            reloadSubSection = true;
+    }
+
+    public bool CanReload()
+    {
+        if (!firstLoading)
+            return reloadSubSection && !IsInCooldown();
+
+        firstLoading = false;
+        reloadSubSection = true;
+        lastTimeAPIChecked = Time.realtimeSinceStartup - PlacesAndEventsSectionComponentController.MIN_TIME_TO_CHECK_API;
+
+        return true;
+
+        bool IsInCooldown() => lastTimeAPIChecked < Time.realtimeSinceStartup - PlacesAndEventsSectionComponentController.MIN_TIME_TO_CHECK_API;
+    }
+
+    public void RequestAll()
+    {
+        view.RestartScrollViewPosition();
+        view.SetAllAsLoading();
+
+        reloadSubSection = false;
+        lastTimeAPIChecked = Time.realtimeSinceStartup;
+
+        if (!exploreV2Menu.isInShowAnimationTransiton.Get())
+            requestAllFromAPI();
```

Comment:
cannot this be a method in `IPlacesAndEventsSubSectionComponentView` and avoid passing that obscure Action to the class?

Reply:
I wouldn't like to move this highly non-UI logic (that operates with the Domain model) into the View. What we could do is to pass `IEvents/IPlacesAPIController`s in this class and call it here. But `RequestAllPlacesAndEventsFromAPI ` in the `HighlightsSubSectionComponentController` is really tricky coupled, so refactoring of this part of the system needs more time. 
I would suggest to extract some of your comments in the further refactoring ticket of this SubSections (controllers are still to improve and View's are even worse)

Reply:
As an fast fix, I will pass ref to the Controller and call it from there. It will increase a bit amount of dependencies, but will make the logic a bit clearer

Reply:
I agree with both your comments. Let's mitigate a bit with the reference to the controller and prepare a ticket for further refactoring.